windows/src/developer/TIKE/oskbuilder/UframeTouchLayoutBuilder.pas

```diff
@@ -588,4 +638,50 @@ procedure TframeTouchLayoutBuilder.Redo;
   BuilderCommand('redo');
 end;
 
+
+procedure TframeTouchLayoutBuilder.CharMapDragOver(Sender, Source: TObject; X, Y: Integer;
+  State: TDragState; var Accept: Boolean);
+var
+  j: TJSONObject;
+begin
+  cef.cefwp.SetFocus;
+  j := TJSONObject.Create;
+  try
+    j.AddPair('x', TJSONNumber.Create(X));
+    j.AddPair('y', TJSONNumber.Create(Y));
+    j.AddPair('state', GetEnumName(TypeInfo(TDragState), Ord(State)));
+    BuilderCommand('charmapDragOver', j);
+  finally
```

Comment:
> and throw it away if it is outside bounds during drop.

If there's an exception, would `Accept := False;`?

Reply:
If there is an exception, `Accept` is irrelevant because the function never returns -- control drops to a exception handler. In this code, there is a low chance of exception (in theory it could happen in `BuilderCommand`), and the memory leak in the case of exception would have been fairly irrelevant (especially since an exception here would probably result in program termination).

So, the `try`/`finally` pattern here is more of a maintainability model -- ensuring that resources are disposed of through code structuring and clearly modelling resource lifetime. It helps us as the developers to easily pair the allocation/disposal calls. That's pretty much all it's for :)